src/Hosting/WindowsServices/src/WindowsServiceLifetimeHostBuilderExtensions.cs

```diff
@@ -21,11 +27,41 @@ public static class WindowsServiceLifetimeHostBuilderExtensions
         /// <returns></returns>
         public static IHostBuilder UseWindowsService(this IHostBuilder hostBuilder)
         {
+            return hostBuilder.UseWindowsService((context, settings) =>
+            {
+                settings.SourceName = context.HostingEnvironment.ApplicationName;
+            });
+        }
+
+        /// <summary>
+        /// Sets the host lifetime to WindowsServiceLifetime, sets the Content Root,
+        /// and enables logging to the event log.
+        /// </summary>
+        /// <remarks>
+        /// This is context aware and will only activate if it detects the process is running
+        /// as a Windows Service.
+        /// </remarks>
+        /// <param name="hostBuilder"></param>
+        /// <param name="configureEventLog"></param>
+        /// <returns></returns>
+        public static IHostBuilder UseWindowsService(this IHostBuilder hostBuilder, Action<HostBuilderContext, EventLogSettings> configureEventLog)
```

Comment:
If you add IOptions<EventLogSettings> support to EventLogLogger provider you can avoid having this overload and make EventLog logger more consistent with others.